javatests/arcs/android/storage/database/DatabaseImplTest.kt

```diff
@@ -1275,6 +1278,76 @@ class DatabaseImplTest {
             .isEqualTo(collection.copy(values = newValues))
     }
 
+    @Test
+    fun garbageCollection() = runBlockingTest {
+        val schema = newSchema("hash")
+        val backingKey = DummyStorageKey("backing")
+        var version = 1
+        fun entity(id: String, creationDaysAgo: Long) = DatabaseData.Entity(
+            RawEntity(
+                id,
+                singletons = mapOf(),
+                collections = mapOf(),
+                creationTimestamp = JvmTime.currentTimeMillis - Duration.ofDays(creationDaysAgo).toMillis()
+            ),
+            schema,
+            FIRST_VERSION_NUMBER,
+            VERSION_MAP
+        )
+        suspend fun updateCollection(vararg entities: DatabaseData.Entity) {
+            val values = entities.map { Reference(it.rawEntity.id, backingKey, VersionMap("ref" to 1)) }
+            val collection = DatabaseData.Collection(
+                values = values.toSet(),
+                schema = schema,
+                databaseVersion = version++,
+                versionMap = VERSION_MAP
+            )
+            database.insertOrUpdate(DummyStorageKey("collection"), collection)
+        }
+
+        val entityInCollectionKey = DummyStorageKey("backing/entity1")
+        val entityInCollection = entity("entity1", 10)
+        val orphanEntityKey = DummyStorageKey("backing/entity2")
+        val orphanEntity = entity("entity2", 10)
+        val recentEntityKey = DummyStorageKey("backing/entity3")
+        val recentEntity = entity("entity3", 1)
+        val lateRefdEntityKey = DummyStorageKey("backing/entity4")
+        val lateRefdEntity = entity("entity4", 10)
```

Comment:
When it comes to reading logs for this test when it fails, it would be easier to follow-along if the storage keys and IDs matched their intended purpose (the variable names seem like better names).

Something like:
```suggestion
        val entityInCollectionKey = DummyStorageKey("backing/entityInColection")
        val entityInCollection = entity("entityInCollection", 10)
        val orphanEntityKey = DummyStorageKey("backing/orphan")
        val orphanEntity = entity("orphan", 10)
        val recentEntityKey = DummyStorageKey("backing/recent")
        val recentEntity = entity("recent", 1)
        val lateRefdEntityKey = DummyStorageKey("backing/lateRef")
        val lateRefdEntity = entity("lateRef", 10)
```